src/protobuf/message.cr

```diff
@@ -137,6 +137,10 @@ module Protobuf
             @{{field[:name].id}} = (%var{tag}).as({{field[:cast_type]}})
           {% end %}
         {% end %}
+
+        {% for tag, field in FIELDS %}
+          self.{{field[:name].id}} = @{{field[:name].id}}
+        {% end %}
```

Comment:
I'm not sure why a second `{% for .. %}`-block is needed here.

It tried to append the `self.X = @x` to the preceding block
but that gave me:

> The instance variable '@is_extension' is initialized in other 'initialize' methods,
and by not initializing it here it's not clear if the variable is supposed
to be nilable or if this is a mistake.

But well, this version works, and I guess it's actually a bit clearer to read.

Reply:
PS: Ah I suppose adding it to the existing `{% for %}` didn't work because
in the case of a `oneof` the setter will iterate over FIELDS _again_ - and may thereby attempt to access fields before they have been initialized. So yep, that makes sense.